src/History/inc/History.h

```diff
@@ -0,0 +1,79 @@
+////////////////////////////////////////////////////////////////////////////
+/**
+ *  @file   History.h
+ *  @date   Tue, June 22 2021
+ *  @brief  Singleton containing the history
+ */
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef CALCULATOR_HISTORY_H
+#define CALCULATOR_HISTORY_H
+
+#include <vector>
+
+#include "calculator_generated.h"
+#include "flatbuffers/flatbuffers.h"
+#include "history_generated.h"
+
+#include "ICalculator.h"
+#include "TermCalculator.h"
+
+namespace calculator {
+    class History
+    {
+        public:
+            /**
+             * @brief delete copy constructors
+             */
+            History(History const&)        = delete;
+            void operator=(History const&) = delete;
+
+            ~History() = default;
+
+            /**
+             * @brief creates a flat buffer object from ICalculator and appends it to calculators_
+             * @param calculator [in]
+             */
+            void appendCalculator(const ICalculator& calculator);
+
+            /**
+             * @brief exposes calculators_.clear()
+             */
+            void clear();
+
+            /**
+             * @brief Get flatbuffer object of the history containing all calculators
+             * @return faltbuffer history object
+             */
+            schema::HistoryT *getBuiltHistory();
+
+            /**
+             * @brief Get the flatbuffer calculator object from calculators_, in
+             * @param index [in]
+             * @return calculatorT if index is valid, otherwise returns nullptr
+             */
+            const schema::CalculatorT *getCalculator(int index);
+
+            /**
+             * @brief returns the Singleton instance, will construct History if needed
+             * @return History singleton instance
+             */
+            static History *getInstance();
+
+            /**
+             * @brief exposes calculators_.size()
+             * @return int - size of calculators_
+             */
+            int getSize();
+        private:
+            static History *instance_;
+            std::vector<const schema::CalculatorT*> calculators_;
```

Comment:
sort and ensure the spaces are correct
std::vector<const schema::CalculatorT*> calculators_;
static History                                           *instance_;

Also, why do we need a private static element?

Reply:
I implemented History as a Singleton, and this seems to be the standard way of implementing a singleton in C++. (from a design perspective might not be the best choice, but I wanted to see how it worked in C++)